File: Modulo3-ObjectiveC-Lorena/Exercicio16.m
//
//  main.m
//  Exercicio16-C
//
//  Created by Lorena Rodrigues Bruno on 06/09/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

#import <Foundation/Foundation.h>

int encontraMaiorElemento(NSMutableArray *array ) {
    int maiorElemento = [array[0] intValue];
    NSInteger tamanhoVetor = array.count;
    int indexArray, indexMaiorElemento;
    
    if (tamanhoVetor > 2) {
        for(indexArray=0;indexArray<=tamanhoVetor-2; indexArray++) {
            if (array[indexArray] > array[indexArray+1] && [array[indexArray] intValue] >= maiorElemento)
            {
                maiorElemento = [array[indexArray] intValue];
                indexMaiorElemento = indexArray;
            }
            if (indexArray == tamanhoVetor-2) {
                if ([array[tamanhoVetor-1] intValue] >= maiorElemento) {
                    maiorElemento = [array[tamanhoVetor-1] intValue];
                    indexMaiorElemento = tamanhoVetor - 1;
                }
            }
        }
    } else if (tamanhoVetor == 2 ){
        if (array[0] > array[1]) {
            maiorElemento = [array[0] intValue];
            indexMaiorElemento = 0;
            
        } else {
            maiorElemento = [array[1] intValue];
            indexMaiorElemento = 1;
        }
        
    } else if (tamanhoVetor == 1) {
        maiorElemento = [array[0] intValue];
        indexMaiorElemento = 0;
    }
    return maiorElemento;
}

int encontraIndexMaiorElemento(NSMutableArray *array, int maiorElemento) {
    NSInteger tamanhoVetor = array.count;
    int i, contador;
    for (i=0; i<tamanhoVetor; i++) {
        if([array[i] intValue] == maiorElemento) {
            contador = i;
            return contador;
        }
    }
    return 0;
}


int main(int argc, const char * argv[]) {
    @autoreleasepool {
        int tamanhoVetor, indexMaiorElemento, maiorElemento, indexArrayOrdenado=0;
        NSMutableArray *array = [@[@14, @7, @9, @12, @13]mutableCopy] ;
        //NSMutableArray *arrayOrdenado = [@[@0, @0, @0, @0, @0]mutableCopy] ;
        int arrayOrdenado[] = {0, 0, 0, 0, 0};
        do {
            maiorElemento = encontraMaiorElemento(array);
            indexMaiorElemento = encontraIndexMaiorElemento(array, maiorElemento);
            arrayOrdenado[indexArrayOrdenado] = maiorElemento;
            [array removeObjectAtIndex:indexMaiorElemento];
            tamanhoVetor = array.count;
            indexArrayOrdenado += 1;
        } while (tamanhoVetor != 0);
        
        printf("Array ordenado %i %i %i %i %i\n", arrayOrdenado[0], arrayOrdenado[1], arrayOrdenado[2], arrayOrdenado[3], arrayOrdenado[4]);

    }
    return 0;
}

File: Modulo3-Lorena/Exercicio16.swift
//
//  main.swift
//  Exercicio16
//
//  Created by Lorena Rodrigues Bruno on 27/08/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

import Foundation

var array = [20, 3, 45, 12, 24]
var tamanhoVetor = array.count
var arrayOrdenado =  Array(repeating: 0, count: tamanhoVetor)
var maiorElemento = Int(), indexMaiorElemento = Int()
var indexArrayOrdenado = Int()

func encontraMaiorElemento(array: [Int]) {
    maiorElemento = array[0]
    if tamanhoVetor > 2 {
        for indexArray in 0...tamanhoVetor-2{
            if array[indexArray] > array[indexArray+1] && array[indexArray] >= maiorElemento
            {
                maiorElemento = array[indexArray]
                indexMaiorElemento = indexArray
            }
            if indexArray == tamanhoVetor-2 {
                if array[tamanhoVetor-1] >= maiorElemento {
                    maiorElemento = array[tamanhoVetor-1]
                    indexMaiorElemento = tamanhoVetor-1
                }
            }
        }
    } else if tamanhoVetor == 2 {
        if array[0] > array[1] {
            maiorElemento = array[0]
            indexMaiorElemento = 0
            
        } else {
            maiorElemento = array[1]
            indexMaiorElemento = 1
        }
        
    } else if tamanhoVetor == 1 {
        maiorElemento = array[0]
        indexMaiorElemento = 0
    }
}


repeat {
    encontraMaiorElemento(array: array)
    arrayOrdenado[indexArrayOrdenado] = maiorElemento
    array.remove(at: indexMaiorElemento)
    print("maior elemento: \(maiorElemento)")
    tamanhoVetor = array.count
    print("tamanho do array: \(tamanhoVetor)")
    indexArrayOrdenado += 1
    
    
} while tamanhoVetor != 0


print("Array ordenado \(arrayOrdenado[0]) \(arrayOrdenado[1]) \(arrayOrdenado[2]) \(arrayOrdenado[3]) \(arrayOrdenado[4])")

